apitools/base/py/transfer.py

```diff
@@ -339,12 +341,18 @@ def __SetRangeHeader(self, request, start, end=None):
         else:
             request.headers['range'] = 'bytes=%d-%d' % (start, end)
 
-    def __GetChunk(self, start, end=None, additional_headers=None):
+    def __GetChunk(self, start, end=None, additional_headers=None,
+                   ignore_chunksize=False):
         """Retrieve a chunk, and return the full response."""
         self.EnsureInitialized()
         end_byte = end
-        if self.total_size and end:
-            end_byte = min(end, self.total_size)
+        if not ignore_chunksize:
```

Comment:
It feels a bit strange to have a function named GetChunk care about chunking.  Why not push the logic around "end" into a helper function that's called by StreamMedia + GetRange?